src/compiler/checker.ts

```diff
@@ -20331,6 +19958,75 @@ namespace ts {
             return resolveCall(node, callSignatures, candidatesOutArray, isForSignatureHelp, headMessage);
         }
 
+        function createSignatureForJSXIntrinsic(node: JsxOpeningLikeElement, result: Type): Signature {
+            // TODO: Since this builds some nodes for the fake'd up signature, this might be worth caching on the node
+            const namespace = getJsxNamespaceAt(node);
+            const exports = namespace && getExportsOfSymbol(namespace);
+            // TODO: We fake up a SFC signature for each intrinsic, however a more specific per-element signature drawn from the JSX declaration

```

Comment:
I would decide what to do about these TODOs and remove them. (I suspect the best thing to do for the second one is just to note that the more-specific signature would be preferable.)

Reply:
The TODOs are here to gather some remarks on what y'all think is appropriate. If I had a solid stance I'd have done it. See, I don't know if the nodes _need_ to be cached, since the _signature_ is already cached when this is called via `resolveSignature` and since that signature should be the only handle to those generated nodes, it might not be necessary.